docs/src/ttmlir-translate.md

```diff
@@ -0,0 +1,12 @@
+# `ttmlir-translate`
+
+The `ttmlir-translate` translation utility. Unlike `ttmlir-opt` tool which is used to run passes within the MLIR world, `ttmlir-translate` allows us to ingest something (e.g. code) into MLIR world, and also produce something (e.g. executable binary, or even code again) from MLIR.
+## Generate C++ code from MLIR
+
+```bash
+# First, let's run `ttmlir-opt` to convert to proper dialect
+./build/bin/ttmlir-opt --ttir-layout --ttnn-open-device --convert-ttir-to-ttnn --convert-ttnn-to-emitc test/ttmlir/Dialect/TTNN/simple_multiply.mlir -o c.mlir
+
+# Now run `ttmlir-translate` to produce C++ code
+./build/bin/ttmlir-translate -mlir-to-cpp c.mlir -allow-unregistered-dialect
```

Comment:
Why do we need `-allow-unregistered-dialect`?  Seems like we should explicitly register the necessary dialects so this flag doesn't need to be provided.

Reply:
It's only because of the top-level module's attribute, `tt.system_desc`:

```
module attributes {torch.debug_module_name = "_lambda", tt.system_desc = #tt.system_desc<[{arch = <wormhole_b0>, grid = <8x8>, l1_size = 1048576, num_dram_channels = 12, dram_channel_size = 1048576}], [0], [<pcie|host_mmio>], [<0, 0, 0, 0>]>}
```

I plan on handling that in one of the following contributions...